### PR TITLE
Update performanceProfile API JSON with corresponding YAML

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR includes following changes:

- Add required new fields in `performanceProfile` for correspondence with [YAML](https://github.com/kruize/autotune/blob/mvp_demo/manifests/autotune/performance-profiles/resource_optimization_local_monitoring.yaml) file
- Update `/createPerformanceProfile` and `/listPerformanceProfile` APIs to include new fields: `apiVersion`, `kind` and `metadata`

### Type of change

- [ ] Bug fix
- [x] New feature
- [X] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [X] Requires DB changes
- [ ] New Tests
- [ ] Updates to existing tests

## How has this been tested?

**Test Configuration**
* Kubernetes clusters tested on: minikube

## Checklist :dart:

- [X] Followed coding guidelines
- [X] Comments added
- [X] Dependent changes merged
- [X] Documentation updated
- [ ] Tests added or updated

## Additional information
docker image - `quay.io/shbirada/perf-profile:json`

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileValidation.java [hunk 1]
```diff
@@ -48,7 +49,9 @@ public class PerformanceProfileValidation {
 
     //Mandatory fields
     private final List<String> mandatoryFields = new ArrayList<>(Arrays.asList(
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileValidation.java:51]: Name is not mandatory ?

**Comment by msvinaykumar** [comment]
please modify MD file

**Comment by shreyabiradar07** [comment]
> please modify MD file

Updated the performance profile documentation

**Comment by msvinaykumar** [comment]
In today's call Dino requested to provide backward compatibility support for this PR.

**Comment by msvinaykumar** [comment]
Certainly! Here's a review checklist you can add as a comment in a pull request:

---

### Review Checklist

1. **Common Review**
2. **Logical Flow Review**
3. **Demo Update**
4. **Naming Conventions**
5. **Workflow**
6. **Does it affect ROS?**

---

**Comment by chandrams** [comment]
@msvinaykumar - You can add the missing ones to the checklist in the pull request template -  https://github.com/kruize/autotune/blob/master/.github/pull_request_template.md

**Comment by shreyabiradar07** [comment]
Closing #1205 as per the design discussion to create new Metric Profile APIs - #1224

**Pull request closed without merging**